### ISSUE TITLE
How to prevent conflicts with another plugin?

### ISSUE THREAD

**Issue by tangrufus** [A5hleyRich/wp-queue#2]
Specifically when installed via composer, should we (as plugin authors):
 - change namespace (via [imposter](https://github.com/TypistTech/imposter) / [mozart](https://github.com/coenjacobs/mozart))?

 - change table names so other plugins won't try to drop my tables?
```php
$this->jobs_table     = $this->database->prefix . 'queue_jobs' . '_my_plugin';
$this->failures_table = $this->database->prefix . 'queue_failures' . '_my_plugin';
```

- change filter and hook names?
```php
apply_filters( 'my_plugin_wp_queue_cron_time_limit', 20 );
apply_filters( 'my_plugin_wp_queue_connections', $connections );
apply_filters( 'my_plugin_wp_queue_cron_memory_exceeded', $return );
```

**Comment by szepeviktor** [comment]
@TangRufus Seems like those are fixed.

**Comment by dingo-d** [comment]
Only partially if you are using imposter plugin https://github.com/deliciousbrains/wp-queue/issues/15.

**Comment by ianmjones** [comment]
> Specifically when installed via composer, should we (as plugin authors):
> 
> * change namespace (via [imposter](https://github.com/TypistTech/imposter) / [mozart](https://github.com/coenjacobs/mozart))?

Yes, although we'd recommend [PHP-Scoper](https://github.com/humbug/php-scoper) these days.

> * change table names so other plugins won't try to drop my tables?

Yes, you should definitely use unique custom table names.
  
> * change filter and hook names?

Yes, in your subclasses override the functions that use `apply_filters()` and rename them to be unique to your plugin.

This last one in particular is a bit of a pain, we should probably make this easier with some sort of prefix property or similar, and maybe make it easier to install the two tables with the same prefix included in their names.

I've marked this issue as an enhancement that we should look into.
